app/frontend/src/components/admin/ServiceClients.vue

```diff
@@ -0,0 +1,231 @@
+    </v-toolbar>
+    <v-container>
+      <!-- search input -->
+      <div class="kc-search">
+        <v-text-field
+          v-model="search"
+          append-icon="mdi-magnify"
+          label="Search"
+          single-line
+          hide-details
+          class="pb-5"
+        ></v-text-field>
+      </div>
+      <!-- table alert -->
+      <v-alert v-if="showAlert" :type="alertType" tile dense>{{alertMessage}}</v-alert>
+      <!-- table header -->
+      <v-data-table
+        class="kc-table"
+        :headers="headers"
+        :items="serviceClients"
+        :items-per-page="10"
+        :search="search"
+        :loading="loading"
+        loading-text="Loading... Please wait"
+        :expanded.sync="expanded"
+        item-key="acronym"
+        show-expand
+      >
+        <!-- environments -->
+        <template v-slot:item.dev="{item}">
+          <div v-if="item.dev">
+            <v-tooltip top>
+              <template v-slot:activator="{ on }">
+                <v-icon color="green" v-on="on">check</v-icon>
+              </template>
+              <span>updated: {{ formatEnvDate(item.environments.DEV) }}</span>
+            </v-tooltip>
+          </div>
+        </template>
+        <template v-slot:item.test="{item}">
+          <div v-if="item.test">
+            <v-tooltip top>
+              <template v-slot:activator="{ on }">
+                <v-icon color="green" v-on="on">check</v-icon>
+              </template>
+              <span>updated: {{ formatEnvDate(item.environments.TEST) }}</span>
+            </v-tooltip>
+          </div>
+        </template>
+        <template v-slot:item.prod="{item}">
+          <div v-if="item.prod">
+            <v-tooltip top>
+              <template v-slot:activator="{ on }">
+                <v-icon color="green" v-on="on">check</v-icon>
+              </template>
+              <span>Updated: {{ formatEnvDate(item.environments.PROD) }}</span>
+            </v-tooltip>
+          </div>
+        </template>
+
+        <!-- expanded row -->
+        <template v-slot:expanded-item="{ headers, item }">
+          <td :class="[responsiveCell]"></td>
+          <td :colspan="headers.length - 1" :class="[responsiveCell]">
+            <div class="kc-expanded">
+              <!-- app details -->
+              <strong>Name: </strong>
+              <span>{{ item.name }}</span>
+            </div>
+
+            <div class="kc-expanded">
+              <!-- app details -->
+              <strong>Created: </strong>
+              <span>{{ formatDate(item.acronymDetails.createdAt) }}</span>
+            </div>
+
+            <div class="kc-expanded">
+              <!-- users -->
+              <strong>Users: </strong>
+              <ul>
+                <li
+                  v-for="(user, index) in item.users"
+                  :key="index"
+                >{{ user.user.firstName + ' ' + user.user.lastName + ((item.users.length > (index + 1)) ? ',' : '') }}</li>
```

Comment:
You probably don't need that many parentheses for evaluating where in the array it is - this expression would be equivalent and suffice:
`(index <= item.users.length) ? ',' : ''`

Reply:
Your code wouldnt work for showing commas.
this would work:
`( (index + 1)  < item.users.length ) ? ',' : ''
`
but its basically what i have

Reply:
Let's just run with that then. Either way, generally it's better to do conditionals with the first term being the value to check, and the second term being the constant you're comparing against (which is what your latest example has. I'd still remove the inner-most parentheses though:
```
( index + 1 < item.users.length ) ? ',' : ''
```